pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test: TestSQLStatsCompactor failed

pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test.TestSQLStatsCompactor [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12067157?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12067157?buildTab=artifacts#/) on release-22.2 @ [9ef28a4293319f74004fcd89fa156b95c903d36d](https://github.com/cockroachdb/cockroach/commits/9ef28a4293319f74004fcd89fa156b95c903d36d):


```
=== RUN   TestSQLStatsCompactor
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor3176569646
    test_log_scope.go:79: use -show-logs to present logs inline
*
* INFO: Running test with the default test tenant. If you are only seeing a test case failure when this message appears, there may be a problem with your test case running within tenants.
*
=== CONT  TestSQLStatsCompactor
    compaction_test.go:268: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/9a30d5271a65be6b3f2aa1433be9011f/logTestSQLStatsCompactor3176569646
--- FAIL: TestSQLStatsCompactor (74.07s)
=== RUN   TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0
    compaction_test.go:235: 
        	Error Trace:	pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats_test_test/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go:235
        	Error:      	Not equal: 
        	            	expected: 28
        	            	actual  : 20
        	Test:       	TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0
        	Messages:   	expected 28 number of wide scans issued, but 20 number of wide scan issued
    --- FAIL: TestSQLStatsCompactor/stmtCount=200/maxPersistedRowLimit=205/rowsDeletePerTxn=0 (13.83s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-observability
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSQLStatsCompactor.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-32095

--- a/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go
+++ b/pkg/sql/sqlstats/persistedsqlstats/compaction_test.go
@@ -155,6 +155,9 @@ func TestSQLStatsCompactor(t *testing.T) {
 			// Test creates a new compactor and calls it directly.
 			sqlConn.Exec(t, "SET CLUSTER SETTING sql.stats.cleanup.recurrence = '@yearly';")
 
+			// Disable auto split by load as it affects the number of wide scans.
+			sqlConn.Exec(t, "SET CLUSTER SETTING kv.range_split.by_load_enabled = false")
+
 			_, err := internalExecutor.ExecEx(
 				ctx,
 				"truncate-stmt-stats",

